pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1587,7 +1587,11 @@ func (cc *XController) deleteQueueJob(obj interface{}) {
 		accessor.SetDeletionTimestamp(&current_ts)
 	}
 	klog.V(3).Infof("[Informer-deleteQJ] %s enqueue deletion, deletion ts = %v", qj.Name, qj.GetDeletionTimestamp())
-	cc.enqueue(qj)
+	//Remove stale copy
```

Comment:
Does this mean the AppWrapper wasn't actually deleted? There isn't any `Delete` calls inside this `deleteQueueJob` functions.

Reply:
We want to keep AW inside MCAD internal queue that has the latest information about deletion timestamp. If the user does oc apply and then later does oc delete we want AW with deletion timestamp to be present inside the MCAD queue. most CRUD methods on MCAD's internal queue does `addifnotpresent` operation but there is no deep semantic check for object equality

Reply:
Without a finalizer, we are not likely to see an AppWrapper with a deletion timestamp. Do we observe these?

Reply:
MCAD sets deletion timestamp here https://github.com/project-codeflare/multi-cluster-app-dispatcher/blob/4d2c3eee5e8dffb09faf38f65b8f3a3a4ea00de2/pkg/controller/queuejob/queuejob_controller_ex.go#L1585

Reply:
Oh these are MCAD timestamps not kube timestamps. OK.

Reply:
I think I understand. This is "imitating" the event queue like the way the controller runtime does. On the next time around when we check the queue, then it will go through a different path that does have the deletion.